web/src/recent_view_ui.ts

```diff
@@ -44,8 +52,13 @@ import * as user_status from "./user_status";
 import * as user_topics from "./user_topics";
 import * as views_util from "./views_util";
 
-let topics_widget;
-let filters_dropdown_widget;
+type Row = {
+    last_msg_id: number;
+    participated: boolean;
+    type: "private" | "stream";
+};
+let topics_widget: ListWidget<ConversationData, Row> | undefined;
+let filters_dropdown_widget: dropdown_widget.DropdownWidget;
```

Comment:
Sigh yeah I don't like it, but because `clear_for_tests` resets `topics_widget`, it has to be more explicitly allowed to be `undefined`. I'm not actually sure why Typescript allows typing `filters_dropdown_widget` as non-undefined since it isn't initialized right away. But that's how we ended up here.